packages/core/src/browser/tree/tree-model.ts

```diff
@@ -68,6 +68,11 @@ export interface TreeModel extends Tree, TreeSelectionService, TreeExpansionServ
      */
     readonly onOpenNode: Event<Readonly<TreeNode>>;
 
+    /**
+     * Event when a node selected by navigator.
+     */
+    readonly onSelectedByEditor: Event<void>;
```

Comment:
I'm not sure that `core` should know about `editors`, maybe we want it to live in the `file-tree-model`.
Else, I think the changes can be made in the actual navigator tree and not the base `tree-widget` for `scrollToAlignment` to use `center` when selecting an editor.

Reply:
Agreed. It may be desirable to locate logic that permits customization of scroll-to behavior in `core`, but the specific `selectedByEditor` logic definitely belongs in the explorer.

Reply:
How about putting `onSelectedByEditor` in `FileNavigatorModel`?

Reply:
> Agreed. It may be desirable to locate logic that permits customization of scroll-to behavior in `core`, but the specific `selectedByEditor` logic definitely belongs in the explorer.

How about putting `onSelectedByEditor` in `FileNavigatorModel`?

Reply:
> I'm not sure that `core` should know about `editors`, maybe we want it to live in the `file-tree-model`.
> Else, I think the changes can be made in the actual navigator tree and not the base `tree-widget` for `scrollToAlignment` to use `center` when selecting an editor.

I'm afraid change `scrollToAlignment`  have to be in base `tree-widget`,  'scrollToAlignment' is a parameter of list in react-virtualized. I don't think of any other way for the time being.😓

Reply:
> How about putting `onSelectedByEditor` in `FileNavigatorModel`?

I'd be fine with moving `onSelectedByEditor` or `onSelectedByResource` to the `FileNavigatorModel` since core should not know about editors, and the change is specific to file-trees. In addition, `scrollToAlignment` is fine to remain in core as it is useful to all trees.

Reply:
> file-trees

`scrollToAlignment` remain in core,  and it needs to calculate by taking `onSelectedByEditor` event. If `onSelectedByEditor` event is not in core，how do we get this event in `tree-widget`,  it will cause circular references🤯

Reply:
One strategy would be to add information to the selected event indicating to the tree how it should scroll to the selected node. Another approach would be to keep the emitter here, but change its name to something like `onScrollToNodeRequested` that fires an event with one of the values allowed for the `scrollToAllignment` (`'auto' | 'end' | 'start' | 'center' | undefined`).